### PR TITLE
Hibernate reactive is Java 11 minimum

### LINKED ISSUE #1317: Hibernate Reactive is JDK 11 minimum
We have to add that check to the validator

### PR DESCRIPTION
**Pull request by timyates**
Closes #1317

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/database/HibernateReactiveFeatureValidator.java [hunk 1]
```diff
@@ -58,7 +59,11 @@ public void validatePreProcessing(Options options, ApplicationType applicationTy
 
     @Override
     public void validatePostProcessing(Options options, ApplicationType applicationType, Set<Feature> features) {
-        // Do nothing
+        if (features.stream().anyMatch(HibernateReactiveFeature.class::isInstance)
+                && options.getJavaVersion().majorVersion() < JdkVersion.JDK_11.majorVersion()) {
+            System.out.println(options.getJavaVersion().majorVersion());
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/database/HibernateReactiveFeatureValidator.java]: remove println statement

File: starter-core/src/test/groovy/io/micronaut/starter/feature/database/DataHibernateReactiveSpec.groovy [hunk 2]
```diff
@@ -36,9 +37,40 @@ class DataHibernateReactiveSpec extends ApplicationContextSpec {
...
 
+    void "data hibernate reactive requires java 11"() {
+        when:
+        new BuildBuilder(beanContext, BuildTool.GRADLE)
+                .jdkVersion(JdkVersion.JDK_8)
+                .features([DataHibernateReactive.NAME, MySQL.NAME])
+                .render()
+
+        then:
+        def ex = thrown(IllegalArgumentException)
```
Comment (sdelamo) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/database/DataHibernateReactiveSpec.groovy]: Much nicer: 

`IllegalArgumentException ex = thrown()`

than: 

`def ex = thrown(IllegalArgumentException)`

File: starter-core/src/test/groovy/io/micronaut/starter/feature/database/HibernateReactiveJpaSpec.groovy [hunk 3]
```diff
@@ -36,10 +37,41 @@ class HibernateReactiveJpaSpec extends ApplicationContextSpec {
...
 
+    void "data hibernate reactive requires java 11"() {
+        when:
+        new BuildBuilder(beanContext, BuildTool.GRADLE)
+                .jdkVersion(JdkVersion.JDK_8)
+                .features([HibernateReactiveJpa.NAME, MySQL.NAME])
+                .render()
+
+        then:
+        def ex = thrown(IllegalArgumentException)
```
Comment (sdelamo) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/database/HibernateReactiveJpaSpec.groovy]: Much nicer: 

`IllegalArgumentException ex = thrown()`
        
than: 

`def ex = thrown(IllegalArgumentException)`

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Comment by sdelamo** [comment]
I think you have to annotate `DataHibernateReactiveSpec` and `HibernateReactiveJpaSpec` with `@IgnoreIf({ jvm.java8 })`

**Comment by timyates** [comment]
> I think you have to annotate `DataHibernateReactiveSpec` and `HibernateReactiveJpaSpec` with `@IgnoreIf({ jvm.java8 })`

Gah, I had this previously, but it passed locally without it on java 8, so I dropped it...

Added back in 👍

**Comment by timyates** [comment]
@sdelamo Any ideas on the failure?  The AmazonCorreto spec is failing

https://ge.micronaut.io/s/zpd3htqttl3xa/tests/overview?outcome=failed

And seems to be failing quite a lot today

https://ge.micronaut.io/scans/failures?failures.failureClassification=all_failures&failures.failureMessage=Failed%20to%20execute%20goal%20org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile%20(default-compile)%20on%20project%20foo:%20Fatal%20error%20compiling&search.timeZoneId=Europe/London

**Comment by sdelamo** [comment]
> @sdelamo Any ideas on the failure? The AmazonCorreto spec is failing
> 
> https://ge.micronaut.io/s/zpd3htqttl3xa/tests/overview?outcome=failed
> 
> And seems to be failing quite a lot today
> 
> https://ge.micronaut.io/scans/failures?failures.failureClassification=all_failures&failures.failureMessage=Failed%20to%20execute%20goal%20org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile%20(default-compile)%20on%20project%20foo:%20Fatal%20error%20compiling&search.timeZoneId=Europe/London

Restarted the build and it passed 🤷

**Review by sdelamo — APPROVED** [review]

**Pull request merged**